.github/workflows/check.yaml

```
@@ -5,20 +5,55 @@ on: push
 jobs:
   check:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.7", "3.8", "3.9", "3.10"]
+
     steps:
       - uses: actions/checkout@v2
       - uses: actions/setup-python@v2
         with:
-          python-version: "3.10"
+          python-version: ${{ matrix.python-version }}
       - name: Run image
         uses: abatilo/actions-poetry@v2.0.0
         with:
-          poetry-version: "1.1.12"
+          poetry-version: "1.1.13"
+      - run: poetry config virtualenvs.in-project true
+      - name: Set up cache
+        uses: actions/cache@v2
+        id: cache
+        with:
+          path: .venv
+          key: venv-${{ runner.os }}-${{ steps.full-python-version.outputs.version }}-${{ hashFiles('**/poetry.lock') }}
+      - name: Ensure cache is healthy
+        if: steps.cache.outputs.cache-hit == 'true'
+        run: timeout 10s poetry run pip --version || rm -rf .venv
+      - run: poetry install
+      - run: poetry run mypy -p basedtyping -p tests
```

shouldnt the python version be included here? something like
```suggestion
      - run: poetry run mypy -p basedtyping -p tests --python-version ${{ matrix.python-version }}
```

---

hmmmmm, yeah

---

actually that's not needed as mypy uses the interpreters version

---

But the config file overwrites that, I'll remove it from the config